hazelcast/src/main/java/com/hazelcast/query/impl/OrderedIndexStore.java

```diff
@@ -193,51 +201,16 @@ public Iterator<IndexKeyEntries> getSqlRecordIteratorBatch(
                 = descending ? recordMap.descendingMap() : recordMap;
         switch (comparison) {
             case LESS:
-                if (descending) {
-                    navigableMap = navigableMap.tailMap(searchedValue, false);
-                } else {
-                    navigableMap = navigableMap.headMap(searchedValue, false);
-                }
-                break;
+                return getSqlRecordIteratorBatch(NULL, false, searchedValue, false, descending);
```

Comment:
what was the rationale of having so many variants of `getSqlRecordIteratorBatch`? Is `tailMap`/`headMap` significantly cheaper than `subMap`? They all use the same implementation, but `null` end makes it slightly cheaper to find `loNode` and `hiNode`. Does it matter?